modules/gltf/src/lib/gltf-utils/gltf-texture-storage.ts

```diff
@@ -0,0 +1,190 @@
+import type {GLTFTextureInfoMetadata, GLTFMeshPrimitive} from '../types/gltf-json-schema';
+
+import {GLTFScenegraph} from '../api/gltf-scenegraph';
+import {getComponentTypeFromArray} from '../gltf-utils/gltf-utils';
+import {getImageData} from '@loaders.gl/images';
+
+/**
+ * Processes data encoded in the texture associated with the primitive. This data will be accessible through the attributes.
+ * @param scenegraph
+ * @param attributeName
+ * @param featureIdTexture
+ * @param featureTextureTable
+ * @param primitive
+ */
+// eslint-disable-next-line max-statements
+export function processPrimitiveTextures(
```

Comment:
This doesn't really seem like a small well defined, self-contained, reusable `util`, this seems like the core logic of the extension. Should it not be part of the extension file instead of being a util?